### DISCUSSION TITLE
Type a Json field as an Array (to help with autocompletion)

### DISCUSSION THREAD

**Discussion by unlocomqx** [prisma/prisma#10299] [Q&A] [answered]
Hi,

I have a `Json` field declared this way: 
```prisma
model Option {
  option_list Json
  ...
```
I'd like to tell the IDE that it's an array
Currently, when I type `option.option_list.` I get no completion for the array type such as map filter and so on..

The `JsonValue` type includes the `JsonArray` but this doesn't have an effect apparently

I tried on vscode and webstorm just to be sure

Thank you

**Comment by austincrim** [comment, marked as answer]
Hey @unlocomqx 👋!

We don't currently support further typing of `Json` fields, but it is definitely on our radar! Take a look at [this issue]() for further discussion and workarounds.

In your case, you could cast `option_list` to be an array if you wanted proper autocomplete:
```ts
const result = await prisma.option.findFirst()
const option_list = result?.option_list as Array<any> // you could pass another type instead of any if you know the type of option_list
```

Let me know if you have any more questions!

**Reply by unlocomqx** [reply]
Thank you! I will go with that

Cheers

**Reply by nemanjahifolks** [reply]
Whats the difference between Json and Json[] types in the schema?

**Reply by janpio** [reply]
One can be any Json you an imagine, and the other can only be a _list_ of any Json you can imagine - so the top level type of the fields in the Json is defined in the case of `Json[]` while in `Json` it is not.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
